nni/compression/pytorch/speedup/jit_translate.py

```diff
@@ -462,178 +164,295 @@ def forward(self, obj):
-    size_list_node = inputs[1].node()
-    scale_list_node = inputs[3].node()
-    size_list = None
-    scale_list = None
-
-    if size_list_node.kind() == 'prim::ListConstruct':
-        size_list = translate_list(inputs[1], speedup)
-    if scale_list_node.kind() == 'prim::ListConstruct':
-        scale_list = translate_list(inputs[3], speedup)
-    return UpsampleModule(size_list, scale_list)
-
-
-def upsample_nearest2d_python(node, speedup):
-    class UpsampleModule(torch.nn.Module):
-        def __init__(self, size_list, scale_list):
-            super(UpsampleModule, self).__init__()
-            self.size_list = size_list
-            self.scale_list = scale_list
-
-        def forward(self, *args):
-            """
-            The first input of args is the target tensor to upsample
-            , the following parameters is useless, because we already
-            get the size_list and the scale_list by parsing the cpp_nodes.
-            """
-            return torch.nn.functional.upsample_nearest(args[0],
-                                                        size=self.size_list, scale_factor=self.scale_list)
-    c_node = node.key_node
-    inputs = list(c_node.inputs())
-    size_list_node = inputs[1].node()
-    scale_list_node = inputs[2].node()
-    size_list = None
-    scale_list = None
+    Attributes
+    ----------
+    func: Callable
+        The function or method to be called.
+    positional: List
+        Positional arguments.
+    keyword: dict
+        Keyword arguments.
+    undetermined: List[int | str]
+        A list of the right positions of arguments.
+        Position is an int in positional or a str in keyword.
+    special_treat: Dict[int | str, Callable]
+        A Dict of the positions and methods.
+        The values of these positions should be treat by those methods.
 
-    if size_list_node.kind() == 'prim::ListConstruct':
-        size_list = translate_list(inputs[1], speedup)
-    if scale_list_node.kind() == 'prim::ListConstruct':
-        scale_list = translate_list(inputs[2], speedup)
-    return UpsampleModule(size_list, scale_list)
+    """
+    def __new__(cls, func: Callable, positional: List[Any], keyword: Dict[str, Any],
+                undetermined: List[int | str], special_treat: Dict[int | str, Callable]):
+        if not callable(func):
+            raise TypeError("the first argument must be callable")
+
+        if hasattr(func, "func"):
+            positional = func.positional + positional
+            keyword = {**func.keyword, **keyword}
+            func = func.func
+
+        self = super(my_partial, cls).__new__(cls)
+
+        self.func = func
+        self.positional = positional
+        self.keyword = keyword
+        self.undetermined = undetermined
+        self.special_treat = special_treat
+        return self
+
+    def __call__(self, /, *args):
+        assert len(args) >= len(self.undetermined)
+        if len(args) > len(self.undetermined):
```

Comment:
the length of `args` must be greater or equal to the length of the `self.undetermined`. or we have no enough arguments to call the function.

but in the ast graph, an op node may have more arguments than the actual if some arguments are lists, because every value in the list may be in the 'inputs' of the op. so we should throw these values away.